spec/api/comment_thread_spec.rb

```diff
@@ -592,7 +592,24 @@ def thread_result(id, params)
         comment.reload
         comment.endorsed.should == false
         comment.endorsement.should == nil
-        check_thread_result_json(nil, changed_thread, parse(last_response.body))
+        json_response = parse(last_response.body)
+        check_thread_result_json(nil, changed_thread, json_response)
+        json_response["read"].should == false
+      end
+      it "update information of comment thread and mark thread as read" do
+        thread = CommentThread.first
+        user = create_test_user(42)
+        put "/api/v1/threads/#{thread.id}", body: "new body", title: "new title", commentable_id: "new_commentable_id", thread_type: "question", read: true, user_id: user.id
+        last_response.should be_ok
+        changed_thread = CommentThread.find(thread.id)
+        changed_thread.body.should == "new body"
+        changed_thread.title.should == "new title"
+        changed_thread.commentable_id.should == "new_commentable_id"
+        changed_thread.thread_type.should == "question"
+        user.mark_as_read(thread)
```

Comment:
@cahrens @robrap 
That was because the user is created within this test and does not get updated with read states (as it is not returned in response of API where read_states are being added). So I replicated the behaviour of mark_as_read from API to test (as it was being used in all GET tests so I assumed we are using it as convention). But you guys are right that this makes confusion so instead of replicating mark_ad_read I have reloaded user to get its read_states.

Updated this PR; kindly review
P.S:
Should we consider changing GET test cases too? If yes, then probably we can create a task in backlog for that.